packages/realm-react/src/__tests__/useQueryRender.test.tsx

```diff
@@ -354,4 +371,34 @@ describe.each`
 
     expect(queryByText("756c")).toBeNull();
   });
+  // This replicates the issue https://github.com/realm/realm-js/issues/4375
+  it("will handle multiple async transactions", async () => {
+    const { queryByTestId } = await setupTest({ queryType, useUseObject: true });
+    const asyncEffect = async () => {
```

Comment:
If this is a well known naming convention then no worries, but I thought this could be named more plainly

```suggestion
    const performTest = async () => {
```